Sources/mpc-core-kit-swift/MpcCoreKitSwift.swift

```diff
@@ -0,0 +1,590 @@
+        var shareIndexes = try tkey.get_shares_indexes()
+        shareIndexes.removeAll(where: { $0 == "1" })
+
+        // TSS Module Initialize - create default tag
+        // generate factor key or use postboxkey hash as factor
+        let factorKey: String
+        let descriptionTypeModule: FactorType
+
+
+        let hashFactorKey = try Utilities.getHashedPrivateKey(postboxKey: postboxKey!, hashedFactorNonce: option.hashedFactorNonce)
+
+        if option.disableHashedFactorKey == false {
+            factorKey = hashFactorKey
+            descriptionTypeModule = FactorType.HashedShare
+
+        } else {
+            // random generate
+            factorKey = try curveSecp256k1.SecretKey().serialize()
+            descriptionTypeModule = FactorType.DeviceShare
+
+            // delete exisiting hashFactor backupshare if available
+            try await deleteMetadataShareBackup(factorKey: hashFactorKey)
+        }
+
+        // derive factor pub
+        let factorPub = try curveSecp256k1.SecretKey(hex: factorKey).toPublic().serialize(compressed: false)
+
+        // use input to create tag tss share
+        let tssIndex = TssShareType.device
+
+        let defaultTag = "default"
+        try await TssModule.create_tagged_tss_share(threshold_key: tkey, tss_tag: defaultTag, deviceTssShare: nil, factorPub: factorPub, deviceTssIndex: tssIndex.rawValue, nodeDetails: nodeDetails, torusUtils: torusUtils)
+
+        try TssModule.backup_share_with_factor_key(threshold_key: tkey, shareIndex: shareIndexes[0], factorKey: factorKey)
+        // record share description
+        let description = createCoreKitFactorDescription(module: descriptionTypeModule, tssIndex: tssIndex, dateAdded: Int(Date().timeIntervalSince1970))
+        let jsonStr = try factorDescriptionToJsonStr(dataObj: description)
+        try await tkey.add_share_description(key: factorPub, description: jsonStr)
+
+        self.factorKey = factorKey
+        deviceMetadataShareIndex = shareIndexes[0]
+
+        metadataPubKey = try tkey.get_key_details().pub_key.getPublicKey(format: .EllipticCompress)
+
+        let selectedTag = try TssModule.get_tss_tag(threshold_key: tkey)
+        let (tssIndexStr, _) = try await TssModule.get_tss_share(threshold_key: tkey, tss_tag: selectedTag, factorKey: factorKey)
+
+        self.tssIndex = tssIndexStr
+
+        // save as device factor if hashfactor is disable
+        if option.disableHashedFactorKey == true {
+            try await setDeviceFactor(factorKey: factorKey)
+        }
+    }
+
+    public func logout() async throws {
+        // TODO: how to clear all state
+        factorKey = nil
+        authSigs = nil
+        sigs = nil
+
+        tkey = nil
+        metadataHostUrl = nil
+        nodeDetails = nil
+        nodeIndexes = nil
+
+        verifier = nil
+        verifierId = nil
+        userInfo = nil
+
+        postboxKey = nil
+        metadataPubKey = nil
+        deviceMetadataShareIndex = nil
```

Comment:
tssShareIndex is missing in state as we have in web

Reply:
state should look like this

  postBoxKey?: string;
  signatures?: string[];
  userInfo?: UserInfo;
  tssShareIndex?: number;
  tssPubKey?: Buffer;
  accountIndex: number;
  factorKey?: BN;